### PR TITLE
feat: integrate language server

### PR DESCRIPTION
**Pull request by g-plane**
- Enabled TypeScript language service.
- Integrated wat language service.

### CODE REVIEW

File: packages/webassembly-playground/src/lsp/index.ts [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+      "global",
+      "memory",
+      "import",
+      "start",
+      "table",
+      "item",
+      "offset",
+      "mut",
+    ],
+    typeKeywords: ["i32", "i64", "f32", "f64", "v128", "funcref", "externref"],
```
Comment (yfzhe) [inline packages/webassembly-playground/src/lsp/index.ts:46]: Does the wat-language-server support `funcref` and `externref` now?

File: packages/webassembly-playground/src/ui/Editor.tsx [hunk 2]
```diff
@@ -3,35 +3,39 @@ import * as monaco from "monaco-editor";
...
+        tabSize: 2,
+        minimap: { enabled: false },
+      });
+      editorRef.current = editor;
+
+      return () => {
+        editor.dispose();
+      };
+    }, [initialContent]);
+
```
Comment (yfzhe) [inline packages/webassembly-playground/src/ui/Editor.tsx:36]: We should update the language of the editor's current model if the `language` prop changed.

File: packages/webassembly-playground/webpack.config.js [hunk 3]
```diff
@@ -77,6 +77,9 @@ const baseConfig = {
       },
     },
   },
+  experiments: {
+    asyncWebAssembly: true,
```
Comment (yfzhe) [inline packages/webassembly-playground/webpack.config.js:81]: Why do we need to set this option? Frankly, I don't know what webpack's `experiments.asyncWebAssembly` does exactly. Quote from webpack's doc:
> Support the new WebAssembly according to the [updated specification](https://github.com/WebAssembly/esm-integration), it makes a WebAssembly module an async module.

But it looks like `wasm-pack` uses just `WebAssembly.instantiateStreaming` or `.instantiate`, which were introduced before wasm's esm-integration proposal.

**Review by yfzhe — COMMENTED** [review]
I have a few questions. But no changes are requested, I'd like to merge this PR.

File: packages/webassembly-playground/src/ui/Editor.tsx (hunk 2 above)
Comment (g-plane) [reply packages/webassembly-playground/src/ui/Editor.tsx:36]: We need eslint-plugin-react-hooks to detect this.

File: packages/webassembly-playground/webpack.config.js (hunk 3 above)
Comment (g-plane) [reply packages/webassembly-playground/webpack.config.js:81]: I'm not sure, but you can try removing this and check if it works.

File: packages/webassembly-playground/src/lsp/index.ts (hunk 1 above)
Comment (g-plane) [reply packages/webassembly-playground/src/lsp/index.ts:46]: Just behaves like number types and vector type, nothing else.

**Pull request merged**